### PR TITLE
fix should_build for cmake build helper

### PR DESCRIPTION
**Pull request by danimtb**
- [x] Refer to the issue that supports this Pull Request: #3451 
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. Also adding a description of the changes in the ``changelog.rst`` file. https://github.com/conan-io/docs

Without tests

### CODE REVIEW

File: conans/client/build/cmake.py [hunk 1]
```diff
@@ -184,6 +184,11 @@ def configure(self, args=None, defs=None, source_dir=None, build_dir=None,
                 self._conanfile.run(command)
 
     def build(self, args=None, build_dir=None, target=None):
+        if not self._conanfile.should_build:
+            return
+        self._build(args, build_dir, target)
+
+    def _build(self, args=None, build_dir=None, target=None):
         if not self._conanfile.should_build:
```
Comment (lasote) [inline conans/client/build/cmake.py]: Remove this in `_build`?

**Comment by lasote** [comment]
> Without tests

The main issue I see so far ;)

File: conans/client/build/cmake.py (hunk 1 above)
Comment (memsharded) [reply conans/client/build/cmake.py]: Exactly that is the point of the PR, this check shouldn't be here.

Comment (memsharded) [reply conans/client/build/cmake.py]: Please add a test that checks this PR is actually fixing this issue.

**Comment by danimtb** [comment]
Added test

**Comment by memsharded** [comment]
Fix tests.

**Pull request merged**